Added Jest test to verify UMD API-forwarding for scheduling package

bvaughn opened the pull request:
Additional automated check to ensure the two APIs stay in-sync.

bvaughn commented:
Sure. Sounds reasonable.

react-linter commented:
<!--
  0 failure: 
  0 warning: 
  
  1 markdown notices
  DangerID: danger-id-default;
-->




  <details>
  <summary>Details of bundled changes.</summary>

  <p>Comparing: 46950a3dfc9e14b5b91ee5823df11883f5d2466e...152d8d624c6e9f3db15ab12df8486f3a74d48d4b</p>


  
## react-scheduler
File | Filesize Diff | Gzip Diff | Prev Size | Current Size | Prev Gzip | Current Gzip | ENV
 ---  |  ---  |  ---  |  ---  |  ---  |  ---  |  ---  |  --- 
react-scheduler.development.js | +24.4% | +23.6% | 15.4 KB | 19.17 KB | 4.65 KB | 5.74 KB | UMD_DEV
**react-scheduler.production.min.js** | **:small_red_triangle:+15.8%** | **:small_red_triangle:+21.0%** | **2.73 KB** | **3.16 KB** | **1.26 KB** | **1.53 KB** | **UMD_PROD**

  </details>
  
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

bvaughn commented:
The future consideration was a good catch. Thanks for bringing that up.

I've copied the two files and renamed them to match our standard pattern. Also updated tests to cover them both, and updated the fixtures file.

bvaughn commented on packages/react-scheduler/src/__tests__/SchedulingUMDBundle-test.js:
```diff
@@ -0,0 +1,41 @@
+    const umdAPIProd = require('../../npm/umd/react-scheduler-tracking.production.min');
+    const cjsAPI = require('../../tracking');
+    const secretAPI = require('react/src/ReactSharedInternals').default;
+    compareAPIS([umdAPIDev, umdAPIProd, cjsAPI, secretAPI.SchedulerTracking]);
```
I feel kind of clever about this test even though it's silly 😁

gaearon commented on packages/react-scheduler/src/__tests__/SchedulingUMDBundle-test.js:
```diff
@@ -16,25 +16,26 @@ describe('Scheduling UMD bundle', () => {
     jest.resetModules();
   });
 
+  function compareAPIS(apis) {
```
Nit: maybe `APIs` 😄 (I don't care really)

gaearon commented on packages/react-scheduler/src/__tests__/SchedulingUMDBundle-test.js:
```diff
@@ -16,25 +16,26 @@ describe('Scheduling UMD bundle', () => {
+  function compareAPIS(apis) {
+    apis = apis.map(api => Object.keys(api).sort());
+    for (let i = 1; i < apis.length; i++) {
+      expect(apis[0]).toEqual(apis[i]);
```
Might be a bit clearer if the "source of truth" API is passed as first argument. Then you can remove the `let i = 1` special case. Also errors would be better if you do

```js
expect(apis[i]).toEqual(nodeAPI);
```

because it's UMD/Secrets API that are *actual* and the original source code that is *expected*.

gaearon commented:
Urrghhh GitHub shows my fresh comments as outdated whyyyy

bvaughn commented:
Ran all test permutations and Flow checks locally since CircleCI/Yarn are still busted.

gaearon commented:
@bvaughn Did you see my comments?

bvaughn commented:
Yikes. no, I didn't see any comments at all. Which ones did I miss?

gaearon commented:
Let me repost lol

gaearon commented on packages/react-scheduler/src/__tests__/SchedulingUMDBundle-test.internal.js:
```diff
@@ -0,0 +1,41 @@
+  function compareAPIS(apis) {
+    apis = apis.map(api => Object.keys(api).sort());
+    for (let i = 1; i < apis.length; i++) {
+      expect(apis[0]).toEqual(apis[i]);
```
Might be a bit clearer if the "source of truth" API is passed as first argument. Then you can remove the `let i = 1` special case. Also errors would be better if you do

```js
expect(apis[i]).toEqual(nodeAPI);
```

because it's UMD/Secrets API that are *actual* and the original source code that is *expected*.

bvaughn commented:
Yuck! They just popped in!

bvaughn commented:
Why is everything half broken today 😅 I will do another commit here in a sec.

gaearon commented:
(By the way this is why I sometimes just use "add comment" instead of "review". I just don't have trust GitHub won't insta-hide them..)

sebmarkbage commented:
I think we should undo the prod/dev split.

The mental model here is that Webpack should ideally include the scheduler in its runtime for *everyone*. So it needs to be tiny and way unopinionated.

gaearon commented:
Should we undo it for CJS too? Should there be an ESM version? I think we'll need a more detailed plan there. After this change it's consistent with other React packages. If we want to make it sufficiently different let's first decide what's the intended layout structure and what UMD bundle will contain.
